File: Exercism Codes/PHP/PigLatin.php
<?php

declare(strict_types=1);

function translate(string $text): string
{
    $result = [];

    foreach (explode(" ", $text) as $x) {
        $sl = 0;

        while (true) {
            $is_vowel = str_contains("aeiou", $x[$sl]);
            $rule_1 = substr($x, $sl, 2) === "yt" || substr($x, $sl, 2) === "xr";
            $rule_4 = (!str_contains("aeiou", $x[$sl - 1]) && $sl - 1 >= 0) && $x[$sl] === "y";

            if ($is_vowel || $rule_1 || $rule_4) {
                break;
            }

            if (substr($x, $sl, 2) === "qu") {
                $sl += 1;
            }

            $sl += 1;
        }

        $result[] = substr($x, $sl, strlen($x) - $sl) . substr($x, 0, $sl) . "ay";
    }

    return implode(" ", $result);
}

File: Exercism Codes/C#/PigLatin.cs
public static class PigLatin
{
    public static string Translate(string word)
    {
        List<string> result = [];

        foreach (string c in word.Split(" "))
        {
            int sliceIdx = 0;
            while (true)
            {
                bool isVowel = "aeiou".Contains(c[sliceIdx]);
                bool rule1 = sliceIdx + 2 <= c.Length && (c.Substring(sliceIdx, 2) == "yt" || c.Substring(sliceIdx, 2) == "xr");
                bool rule4 = (sliceIdx - 1 >= 0 && !"aeiou".Contains(c[sliceIdx - 1])) && c[sliceIdx] == 'y';

                if (isVowel || rule1 || rule4)
                {
                    result.Add(c[sliceIdx..] + c[..sliceIdx] + "ay");
                    break;
                }
                if (c.Substring(sliceIdx, 2) == "qu")
                {
                    sliceIdx++;
                }
                sliceIdx++;
            }
        }

        return string.Join(" ", result);
    }
}